Move candid ui section to an include file. (#31)
dfinity-lab/docs: modules/developers-guide/examples/candid-ui.adoc

== Testing functions in a browser

The canister interface description language—often referred to as Candid or more generally as the IDL—provides a common language for specifying the signature of a canister service.
The language provides a unified way of interacting with canisters in various languages and tools, such as the `+dfx+` command-line interface, JavaScript, and {proglang}.

> Based on the type signature of the actor, Candid provides a web interface that allows you to call canister functions for testing and debugging.

Reviewer: ```suggestion
Based on the type signature of the actor, Candid provides a web interface that allows you to call canister functions for testing and debugging. Note that the Candid web interface is always available even if you don't setup frontend in your project.
```

== Testing functions using a web interface

> The canister interface description language—often referred to as Candid or more generally as the IDL—provides a common syntax for specifying input arguments and return values.

Author: Changed

> The canister interface description language—often referred to as Candid or more generally as the IDL—provides a common syntax for specifying input arguments and return values.

Author: Changed

> The language performs automatic type mapping so you can use the same syntax to interact with the `+dfx+` command-line interface, JavaScript, {proglang}, and other supported languages or tools.

Author: Changed.

> The language can also automatically generate a default web interface that exposes your program's functions for testing and debugging.

Reviewer: ```suggestion
Based on the type signature of the actor, Candid provides a web interface that allows you to call canister functions in a graphical manner.
```

Author: web interface and graphical manner sort of say the same thing, so I restored the reason for using the graphical user interface--testing and debugging.

> The language can also automatically generate a default web interface that exposes your program's functions for testing and debugging.

Author: Replaced where it seems appropriate.

> To use the default web interface to test canister functions:

Reviewer: ```suggestion
To use the Candid web interface to test canister functions:
```

Author: Changed.

. Deploy your project using the `+dfx canister install+` command and copy the canister identifier that begins with the `+ic:+` prefix.
. Open a browser and navigate to the address and port number specified in the `+dfx.json+` configuration file.
+

> By default, the URL uses the localhost address (`+127.0.0.1+`) and port number `+8000+`, so you can navigate to the front-end for the hello_world program using `+127.0.0.1:8000+` for the URL.

Reviewer: ```suggestion
By default, the URL uses the localhost address (`+127.0.0.1+`) and port number `+8000+`. The URL for Candid UI is located under `/candid` with a required `canisterId` parameter.
```

Author: I would like to keep the two parts of the URL separate and keep the example.

. Append the candid keyword the canister identifier after the host address and port number using the following syntax:
+
[source,bash,subs="quotes"]
----
candid?canisterId=ic:canister_identifier

> ----

Reviewer: Remove line 14--19

Author: I'll remove the format sample

+
For example, the full URL should look similar to the following but with the `+ic:_canister_identifier_+` that was returned by the `+dfx canister install+` command:
+
[source,bash,subs="quotes"]
----
http://localhost:8000/candid?canisterId=ic:81DDA04F69F40FEEAC
----
. Review the list of function calls and types defined in the program.

> . Type a value of the appropriate type for a function, then click *Call* to see the result.

Reviewer: ```suggestion
. Type a value of the appropriate type for a function, then click the *Call* or *Query* button to see the result.
```

Author: Changed
